src/v/raft/consensus.cc

```diff
@@ -2049,18 +2049,19 @@ consensus::do_append_entries(append_entries_request&& r) {
           truncate_at);
         _probe->log_truncated();
 
+        _majority_replicated_index = std::min(
```

Comment:
This wasn't there before, why is it needed?

Reply:
we now must allow `_majority_replicated_index` to move back, as we are truncating beyond that offset and we do not want the majority replicated index to be greater than log end offset